### PR TITLE
Fix deploy dynamic list fall back workflow

### LINKED ISSUE #5274: GitHub actions to deploy fallback lists
Creating the issue as a missing part for:
- https://github.com/wp-media/wp-rocket/issues/5038
- https://github.com/wp-media/rucss-backend/issues/5

During the release we need to create a fallback file to make sure that customers have exclusions when there's no connection with our backend.

### PR DESCRIPTION
**Pull request by mostafa-hisham**
## Description

Create a new git hu workflow to update dynamic list fallback JSON file when push to trunk

Fixes #5274 

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## Is the solution different from the one proposed during the grooming?

No

## How Has This Been Tested?

Tested on a separate GitHub repo to check if workflow is working

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings

### CODE REVIEW

**Review by jeawhanlee — COMMENTED** [review]
Isn't this supposed to be using the release trigger `on:release:` and if so, then we can append the concerned steps to the deploy workflow we already have.
What do you think @mostafa-hisham @Tabrisrp

File: .github/workflows/deploy_dynamic_lists_fallback.yml [hunk 1]
```diff
@@ -0,0 +1,46 @@
+# This is a basic workflow to help you get started with Actions
+
+name: CI
```
Comment (remyperona) [inline .github/workflows/deploy_dynamic_lists_fallback.yml]:
```suggestion
name: Dynamic lists fallback update
```

**Review by remyperona — CHANGES_REQUESTED** [review]

File: .github/workflows/deploy_dynamic_lists_fallback.yml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+          touch dynamic-lists.json
+          echo '${{ steps.getDynamicLists.outputs.lists }}' > dynamic-lists.json
+          cat dynamic-lists.json
+
+
+      - name: Commit the file
+        if: ${{ steps.saveDynamicLists.outcome == 'success' }}
+        uses: stefanzweifel/git-auto-commit-action@v4
+        with:
+          commit_message: update dynamic lits
```
Comment (remyperona) [inline .github/workflows/deploy_dynamic_lists_fallback.yml]:
```suggestion
          commit_message: update dynamic lists
```

File: .github/workflows/deploy_dynamic_lists_fallback.yml [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+          commit_user_name: WP-media GitHub Actions Bot
```
Comment (remyperona) [inline .github/workflows/deploy_dynamic_lists_fallback.yml]:
```suggestion
          commit_user_name: WP Media GitHub Actions Bot
```

**Comment by remyperona** [comment]
Since we want the file to be committed to the trunk branch, it makes sense to do it on push to this branch I think

**Review by remyperona — APPROVED** [review]

**Pull request merged**